src/example_report.rs

```diff
@@ -0,0 +1,90 @@
+use std::convert::From;
+
+#[cfg(feature = "use_expectest")]
+use expectest::core::TestResult as ExpectestResult;
+
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub struct Failure {
+    pub message: Option<String>,
+}
+
+impl Failure {
+    pub fn new<S>(message: Option<S>) -> Self
+        where String: From<S>
```

Comment:
To be frank, it's not the same. There is no blanket impl `String : From<T: ToString>` so it you depends on this bound, it will break something.

Where do you use this ?